extensions/interactions/FractionInput/FractionInput.js

```diff
@@ -12,8 +12,75 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+
+
+oppia.directive('oppiaInteractiveFractionInput', [function() {
+  return {
+    restrict: 'E',
+    scope: {
+      onSubmit: '&'
+    },
+    templateUrl: 'interaction/FractionInput',
+    controller: [
+      '$scope', '$attrs', 'FocusManagerService',
+      'fractionInputRulesService', 'FractionObjectFactory',
+      function($scope, $attrs, FocusManagerService,
+        fractionInputRulesService, FractionObjectFactory) {
+        $scope.answer = '';
+        $scope.labelForFocusTarget = $attrs.labelForFocusTarget || null;
+        var requireSimplestForm =
+          $attrs.requireSimplestFormWithValue === 'true';
+        var errorMessage = '';
+        $scope.FRACTION_INPUT_FORM_SCHEMA = {
+          type: 'unicode',
+          ui_config: {}
+        };
+
+        $scope.getWarningText = function() {
+          return errorMessage;
+        }
+
+        $scope.submitAnswer = function(answer) {
+          try {
+            var fraction = FractionObjectFactory.fromRawInputString(
+              answer);
+            if (requireSimplestForm &&
+              !angular.equals(fraction, fraction.convertToSimplestForm())
+            ) {
+              errorMessage =
+                'Please enter answer in it\'s simplest form';
```

Comment:
There shouldn't be an apostrophe in "it's" -- "it's" stands for "it is".

Also, these messages should be written as clearly as possible, otherwise students won't understand it. Perhaps: "Please enter an answer in simplest form (e.g., 1/3 instead of 2/6)."

Similarly for some of the parsing messages -- e.g. "Please enter answer in fraction format (e.g. 5/3 or 1 2/3)" --> "Please enter a valid fraction (e.g., 5/3 or 1 2/3)."

Reply:
Ah, that's a grammatical error I always seem to make! :) Done. Also about the messages, I was wondering if you think the 'Please enter a valid fraction (e.g., 5/3 or 1 2/3)' is misleading in the sense that it may suggest an integer is invalid? If so perhaps I could change it to 'Please enter a valid fraction (e.g., 1, 1/2 or 1 1/2)'.

Reply:
Hm, 1 isn't a fraction really. Perhaps  'Please enter a valid fraction (e.g., 5/3 or 1 2/3) or whole number.'?